pkg/operator/controller/controller.go

```diff
@@ -934,3 +929,45 @@ func (r *ReconcileVMImportConfig) isMutable(obj runtime.Object) bool {
 	}
 	return false
 }
+
+func (r *ReconcileVMImportConfig) customizeControllerConfiguration(namespace string) error {
+	configMap := corev1.ConfigMap{}
+	configMapID := client.ObjectKey{Namespace: namespace, Name: ctrlConfig.ConfigMapName}
+	if err := r.client.Get(context.TODO(), configMapID, &configMap); err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+		if namespacedName := r.getOsMappingConfigMapName(namespace); namespacedName != nil {
+			configMap.Data = make(map[string]string)
+			configMap.Data[ctrlConfig.OsConfigMapNameKey] = namespacedName.Name
+			configMap.Data[ctrlConfig.OsConfigMapNamespaceKey] = namespacedName.Namespace
+		}
+		configMap.Name = configMapID.Name
+		configMap.Namespace = configMapID.Namespace
+		if err = r.client.Create(context.TODO(), &configMap); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *ReconcileVMImportConfig) getOsMappingConfigMapName(namespace string) *types.NamespacedName {
+	var configMapName, configMapNamespace string
+	operatorDeployment := &appsv1.Deployment{}
+	key := client.ObjectKey{Namespace: namespace, Name: "vm-import-operator"}
+	if err := r.client.Get(context.TODO(), key, operatorDeployment); err == nil {
+		operatorEnv := operatorDeployment.Spec.Template.Spec.Containers[0].Env
```

Comment:
just making sure - there is expected to be only single container per operator's deployment or would it be safer relying on the container-name ?

Reply:
You're right - it will be safer to use the name.

Reply:
Fixed.